src/System.Memory/src/System/SpanHelpers.byte.cs

```diff
@@ -95,6 +95,69 @@ public static int IndexOf(ref byte searchSpace, byte value, int length)
             return -1;
         }
 
+        public static int IndexOf(ref byte searchSpace, byte value, int startIndex, int count, int spanLength)
```

Comment:
I do not think it makes sense to replicate the worker method for this - you should use the existing `public static int IndexOf(ref byte searchSpace, byte value, int length)` method.